### PR TITLE
feat: React - Pagination in Table (#41191)

### PR DESCRIPTION
**Pull request by AryanSinghal**

### CODE REVIEW

File: src/pages/Trainee/components/Table.jsx [hunk 1]
```diff
@@ -1,70 +1,173 @@
...
+                          <TableCell align={column.align || 'center'} onClick={() => { onSelect(row) }}>
+                            {(column.format) ? column.format(row[column.field]) : row[column.field]}
+                          </TableCell>
+                        </Fragment>
+                      ))
+                    }
+                    <TableCell>
+                      <Fragment key={id + index + 'edit'}>
+                        <IconButton
+                          onClick={() => { this.handleEditDialogOpen(row) }}
```
Comment (Rahul-Successive) [inline src/pages/Trainee/components/Table.jsx]: Don't hardcode the actions, make it dynamic as well

**Review by Rahul-Successive — CHANGES_REQUESTED** [review]

File: src/pages/Trainee/components/Table.jsx [hunk 2]
```diff
@@ -1,70 +1,173 @@
...
-          </TableBody>
-        </Table>
-      </TableContainer>
-    </div>
-  );
+            </TableBody>
+          </Table>
+          <div align='right'>
+            <span>
+              {page * ROWS_PER_PAGE + 1} - {(page + 1) * ROWS_PER_PAGE} of {count}
```
Comment (Rahul-Successive) [inline src/pages/Trainee/components/Table.jsx]: as per the task, there should be one prop called rows Per Page which you have skipped and directly used the default ROWS_PER_PAGE constant

File: src/pages/Trainee/components/Table.jsx [hunk 3]
```diff
@@ -1,70 +1,173 @@
...
+              onClick={() => { onChangePage(page, 'right') }}
+              disabled={page >= Math.ceil(count / ROWS_PER_PAGE) - 1}
+              aria-label="next page"
+            >
+              <KeyboardArrowRightIcon />
+            </IconButton>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+          </div>
+        </TableContainer>
+        <RemoveDialog
```
Comment (Rahul-Successive) [inline src/pages/Trainee/components/Table.jsx]: The logic for opening and closing the dialogbox and dialogbox should be outside the table component, keep your table component should be as generic as possible

**Review by Rahul-Successive — CHANGES_REQUESTED** [review]

File: src/pages/Trainee/components/Table.jsx (hunk 3 above)
Comment (AryanSinghal) [reply src/pages/Trainee/components/Table.jsx]: ok sir

File: src/pages/Trainee/components/Table.jsx (hunk 2 above)
Comment (AryanSinghal) [reply src/pages/Trainee/components/Table.jsx]: ok sir

File: src/pages/Trainee/components/Table.jsx (hunk 1 above)
Comment (AryanSinghal) [reply src/pages/Trainee/components/Table.jsx]: ok sir

File: src/pages/Trainee/components/Table.jsx [hunk 4]
```diff
@@ -1,70 +1,114 @@
...
-              ))
-            }
-          </TableBody>
-        </Table>
-      </TableContainer>
-    </div>
-  );
+            </TableBody>
+          </Table>
+          <div align='right'>
```
Comment (Rahul-Successive) [inline src/pages/Trainee/components/Table.jsx]: Check the second point of the description in the task.

**Review by Rahul-Successive — CHANGES_REQUESTED** [review]

File: src/pages/Trainee/components/Table.jsx (hunk 4 above)
Comment (AryanSinghal) [reply src/pages/Trainee/components/Table.jsx]: ok sir, I have made the changes

**Pull request merged**